miasm2/analysis/mem.py

```diff
@@ -0,0 +1,1259 @@
+        """Return the static size of this structure, when available (it is the
+        case by default).
+        """
+        # Child classes can set cls._size if their size is not the sum of
+        # their fields
+        if cls._size is None:
+            return sum(a["field"].size() for a in cls._attrs.itervalues())
+        return cls._size
+
+    def get_size(self):
+        """Return the dynamic size of this structure (e.g. the size of an
+        instance). Defaults to sizeof for this base class.
+
+        For example, MemSizedArray defines get_size but not sizeof, as an
+        instance has a fixed size (because it has a fixed length and
+        field_type), but all the instance do not have the same size.
+        """
+        return self.sizeof()
+
+    def get_field_type(self, name):
+        """Return the MemField subclass instance describing field @name."""
+        return self._attrs[name]['field']
+
+    def get_field(self, name):
+        """Get a field value by name.
+
+        Useless most of the time since fields are accessible via self.<name>.
+        """
+        if name not in self._attrs:
+            raise AttributeError("'%s' object has no attribute '%s'"
+                                 % (self.__class__.__name__, name))
+        field = self._attrs[name]["field"]
+        offset = self._attrs[name]["offset"]
+        return field.get(self._vm, self.get_addr() + offset)
+
+    def set_field(self, name, val):
+        """Set a field value by name. @val is the python value corresponding to
+        this field type.
+
+        Useless most of the time since fields are accessible via self.<name>.
+        """
+        if name not in self._attrs:
+            raise AttributeError("'%s' object has no attribute '%s'"
+                                 % (self.__class__.__name__, name))
+        field = self._attrs[name]["field"]
+        offset = self._attrs[name]["offset"]
+        field.set(self._vm, self.get_addr() + offset, val)
+
+    def deref_field(self, name):
+        """Get the MemStruct pointed by <name> field.
+
+        Useless most of the time since fields are accessible via
+        self.deref_<name>.
+        """
+        addr = self.get_field(name)
+        field = self._attrs[name]["field"]
+        assert isinstance(field, Ptr),\
+               "Programming error: field should be a Ptr"
+        return field.deref_get(self._vm, addr)
+
+    def set_deref_field(self, name, val):
+        """Set the MemStruct pointed by <name> field. @val should be of the
+        type of the pointed MemStruct. The field must be a Ptr.
+
+        Useless most of the time since fields are accessible via
+        self.deref_<name>.
+        """
+        addr = self.get_field(name)
+        field = self._attrs[name]["field"]
+        assert isinstance(field, Ptr),\
+               "Programming error: field should be a Ptr"
+        field.deref_set(self._vm, addr, val)
+
+    def memset(self, byte='\x00'):
```

Comment:
We could imagine to fill memory with a pattern, longer than 1 byte (for instance, think to Metasploit pattern_create.rb)